### ISSUE TITLE
Install-import problem on OsX 10.9

### ISSUE THREAD

**Issue by ppuggioni** [pymc-devs/pymc#400]
Hello,

I have installed PyMC on my OsX 10.9, with Xcode 5.0, Enthought python 2.7.3 64-bit

I used:

<pre><code>pip install pymc
</code></pre>

and it terminates the installation with:

<pre><code>Successfully installed pymc
</code></pre>

In the python shell I type
 <pre><code> import pymc 
</code></pre>

and this is what I get:

<pre><code>Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pymc/__init__.py", line 29, in <module>
    from .CommonDeterministics import *
  File "pymc/CommonDeterministics.py", line 21, in <module>
    from .utils import safe_len, stukel_logit, stukel_invlogit, logit, invlogit, value, find_element
  File "pymc/utils.py", line 14, in <module>
    from . import flib
ImportError: dlopen(pymc/flib.so, 2): no suitable image found.  Did find:
    pymc/flib.so: mach-o, but wrong architecture
</code></pre>

Any idea?

PS: if I download from git the pymc3 bleeding version, it works fine. But the 2.3 has this import problem.

Any help would be much appreciated!

Paolo

**Comment by fonnesbeck** [comment]
Hi Paolo,

You have probably built PyMC with an incompatible version of gfortran. Which are you using?

```
gfortran -v
```

**Comment by ppuggioni** [comment]
Hey, thank you for the reply!

<pre><code>
gfortran -v
Using built-in specs.
COLLECT_GCC=gfortran
COLLECT_LTO_WRAPPER=/sw/lib/gcc4.5/libexec/gcc/i386-apple-darwin10.3.1/4.5.0/lto-wrapper
Target: i386-apple-darwin10.3.1
Configured with: ../gcc-4.5.0/configure --prefix=/sw --prefix=/sw/lib/gcc4.5 --mandir=/sw/share/man --infodir=/sw/lib/gcc4.5/info --enable-languages=c,c++,fortran,objc,obj-c++,java --with-gmp=/sw --with-libiconv-prefix=/sw --with-ppl=/sw --with-cloog=/sw --with-mpc=/sw --with-system-zlib --x-includes=/usr/X11R6/include --x-libraries=/usr/X11R6/lib --program-suffix=-fsf-4.5
Thread model: posix
gcc version 4.5.0 (GCC) 
</code></pre>

**Comment by fonnesbeck** [comment]
Yes, you should probably be using 4.2. You can get it from homebrew, if you use that, or use [this one](http://r.research.att.com/tools/gcc-42-5666.3-darwin11.pkg).

**Comment by ppuggioni** [comment]
Thank you!
I installed the 4.2 fortran compiler, but I do not know how to force pip to use that specific version. Do you have any ready trick for it or pointing to some other forums? After spending many hours and many failed attempts, I am a bit demoralised.. if the request is off-topic I understand, I'll keep looking for the solution!

**Comment by jangevaare** [comment]
I am using pymc 3 with OSX 10.9, Enthought and Xcode 5.0 command line tools with no problems. I will try to replicate here...

As an aside after you get this working, this link will be relevant to you:
https://support.enthought.com/entries/22861925-OS-X-10-9-Mavericks-Python-Canopy

**Comment by ppuggioni** [comment]
Unfortunately I had the bad idea to upgrade to Mavericks, too late to go back..
If you can replicate the pymc3 install it would be great! Thank you!

**Comment by jangevaare** [comment]
For installing pymc 2.3, I needed to get gfortran (got it from http://gcc.gnu.org/wiki/GFortranBinaries#MacOS). I didn't need to do any configuration. Here's the information on the gfortran version I used:

```
Justins-MacBook-Pro:~ justin$ gfortran -v
Using built-in specs.
COLLECT_GCC=gfortran
COLLECT_LTO_WRAPPER=/usr/local/gfortran/libexec/gcc/x86_64-apple-darwin13/4.8.2/lto-wrapper
Target: x86_64-apple-darwin13
Configured with: ../gcc-4.8.2/configure --prefix=/usr/local/gfortran --with-gmp=/Users/fx/devel/gcc/deps-static/x86_64 --enable-languages=c,c++,fortran,objc,obj-c++ --build=x86_64-apple-darwin13
Thread model: posix
gcc version 4.8.2 (GCC) 
```

On to installing pymc 2.3

```
Justins-MacBook-Pro:~ justin$ sudo pip install git+https://github.com/pymc-devs/pymc@2.3
```

There were several warnings related to the fortran compiler while installing... probably related to the version I used. But:

```
Successfully installed pymc
Cleaning up...
Justins-MacBook-Pro:~ justin$ python
Enthought Canopy Python 2.7.3 | 64-bit | (default, Aug  8 2013, 05:37:06) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pymc as pm
>>>
```

I then uninstalled 2.3, and reinstalled the latest version of 3.0, which also imports without problem.

```
Justins-MacBook-Pro:~ justin$ sudo pip install --upgrade --no-deps git+https://github.com/pymc-devs/pymc
Downloading/unpacking git+https://github.com/pymc-devs/pymc
  Cloning https://github.com/pymc-devs/pymc to /tmp/pip-zOT8FD-build
  Running setup.py egg_info for package from git+https://github.com/pymc-devs/pymc

Installing collected packages: pymc
  Running setup.py install for pymc

Successfully installed pymc
Cleaning up...
Justins-MacBook-Pro:~ justin$ python
Enthought Canopy Python 2.7.3 | 64-bit | (default, Aug  8 2013, 05:37:06) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pymc as pm
>>> 
```

So... yeah... maybe try other versions of gfortran, such as the one I linked to. I didn't need to do any configuration of pip to have it use the compiler.

If you can't get that to work you could also download the branch locally, navigate to the pymc folder in terminal, and type:

```
python setup.py config
python setup.py build
sudo python setup.py install
```

If you need to modify the compiler options it's probably easier to do it this way.

**Comment by ppuggioni** [comment]
Thank you so so so so much!

Now it works. The reason of the problem was that in my PATH I had /sw/bin/ before the /usr/local/ and so I kept using the old fortran compiler, despite installing the new one. I fixed it and now I am using the gfortran 4.8.2

Both pymc2.3 and pymc3 are correctly imported.

Thank you again :)
